src/gtest/test_keys.cpp

```diff
@@ -235,6 +254,24 @@ TEST(Keys, DeriveUnifiedFullViewingKeys)
                 auto key = libzcash::SaplingDiversifiableFullViewingKey::Read(ss);
                 EXPECT_EQ(key, saplingKey);
             }
+            if (!test[2].isNull()) {
+                auto expectedHex = test[2].get_str();
+
+                // Ensure that the serialized Orchard fvk matches the test data.
+                auto orchardKey = ufvk.GetOrchardKey().value();
+                CDataStream ssEncode(SER_NETWORK, PROTOCOL_VERSION);
+                ssEncode << orchardKey;
+                EXPECT_EQ(ssEncode.size(), 96);
+                auto skeyHex = HexStr(ssEncode.begin(), ssEncode.end());
+                EXPECT_EQ(expectedHex, skeyHex);
+
+                // Ensure that parsing the test data derives the correct dfvk
+                auto data = ParseHex(expectedHex);
+                ASSERT_EQ(data.size(), 96);
+                CDataStream ss(data, SER_NETWORK, PROTOCOL_VERSION);
+                auto key = libzcash::OrchardFullViewingKey::Read(ss);
+                EXPECT_EQ(key, orchardKey);
+            }
             // Enable the following after Orchard keys are supported.
```

Comment:
Should this still be disabled? If it should, add a reference to the relevant issue.

Reply:
This test is checking that constructing a UFVK from the same seed as the test vectors gives the same receivers, but the final overall check cannot be enabled without us having a way to construct a UFVK manually with an unknown component, because the test vectors chose to include an unknown component that by definition we cannot derive. Adding that API is out of scope for this PR.

Reply:
We should modify the test vectors to also generate the UA that contains only the known receiver types.

Reply:
Why not do this check only for the UAs without unknown components?